For main

AleksandrZaec opened the pull request.

AleksandrZaec committed:
добавление ветки, изменил схемы данных team, изменил метод получения детальной информации по группе, аписал механизм добавления в группу участника и выдачи ему роли

AleksandrZaec committed:
написал дополнительную модель для хранения изменений статуса задачи, переделал круд для задач, описал схемы для задач, написал роуты для задач

AleksandrZaec committed:
Merge pull request #1 from AleksandrZaec/tasks

tasks

AleksandrZaec committed:
аписал апи который возвращает список задач связянных с полдьзователем

AleksandrZaec committed:
Merge pull request #3 from AleksandrZaec/tasks_filter

аписал апи который возвращает список задач связянных с полдьзователем

AleksandrZaec committed:
реализовал функционал связанный с комментариями, написал круд, роуты, схемы для комментариев, в модель комментариев добавил геттер для вычисления полного имени пользователя

AleksandrZaec committed:
Merge pull request #4 from AleksandrZaec/comments

реализовал функционал связанный с комментариями, написал круд, роуты,…

AleksandrZaec committed:
добавил эндпоинт для получения задач связанных с группой/командой, в эндпоинт который отдает задачи пользователя добавил фильтр по группам

AleksandrZaec committed:
вернул, метод пост в get_my_tasks,

AleksandrZaec committed:
Merge pull request #5 from AleksandrZaec/revision_tasks

Revision tasks

AleksandrZaec committed:
реализовал функционал по назначению исполнителя задачи, исправил баг в авторизации, при смене токена через рефреш токен выдавался недействительный аксес, сейчас норм.

AleksandrZaec committed:
Merge pull request #7 from AleksandrZaec/task_user

реализовал функционал по назначению исполнителя задачи, исправил баг …

AleksandrZaec committed:
реализовал функционал по оценки выполнения задачи, изменил логику добавлению пользователей в команду, теперь можно добавлять сразу списком, аналогично изменил назначение исполнителей на задачу.

AleksandrZaec committed:
Merge pull request #8 from AleksandrZaec/evaluation

реализовал функционал по оценки выполнения задачи, изменил логику доб…

AleksandrZaec committed:
изменил методы добавления пользователя в группу или к задаче, и методы удаления пользователя из группы и задачи, ко всем, обернул await db.commit() и await db.refresh() в try-except блоки, при ошибках базы данных выполняется откат сессии (rollback)

AleksandrZaec committed:
Merge pull request #9 from AleksandrZaec/add/dell

изменил методы добавления пользователя в группу или к задаче, и метод…

AleksandrZaec committed:
реализовал возможность создавать задачи с пользователями, и создавать группы с пользователями

AleksandrZaec committed:
Merge pull request #10 from AleksandrZaec/refactoring

реализовал возможность создавать задачи с пользователями, и создавать…

AleksandrZaec committed:
Реализовано создание команды и задачи с одновременным добавлением пользователей в одной транзакции.

AleksandrZaec committed:
изменил метод создания команды, иеперь создатель добавляется в команду автоматически

AleksandrZaec committed:
реализовал круд для встреч, и подключил админку

AleksandrZaec committed:
Merge pull request #14 from AleksandrZaec/meeting

Meeting

AleksandrZaec committed:
написал логику для календаря, написал зависимости которые проверяют права на взаимодействие с ресурсом, добавил в роуты краткое и полное описание, что бы понятно отображалась документация

AleksandrZaec committed:
Merge pull request #15 from AleksandrZaec/calendar

написал логику для календаря, написал зависимости которые проверяют п…

AleksandrZaec committed:
написал unit тесты для ф-ций хэширования пароля и создания access и refresh токенов, написал интеграционные тесты для user crud

AleksandrZaec committed:
написал тесты

AleksandrZaec committed:
написал тесты для роутов user, team, task

AleksandrZaec committed:
Merge pull request #16 from AleksandrZaec/tests

Tests

AleksandrZaec committed:
заполнил README.md добавил контейнеризацию

Yasuo20012001 commented on docker-compose.yaml:
```diff
@@ -0,0 +1,63 @@
+volumes:
+  pg_data:
+  pg_test_data:
+
```
1   version: "3.9" — актуальная версия синтаксиса Compose, совместима с большинством возможностей
5   image: postgres:16 — используйте конкретный tag (например, 16.2) для стабильности в CI/CD
9   env_file: — убедитесь, что `.env` добавлен в `.gitignore`, чтобы не хранить секреты в Git
11  healthcheck: — хорошая практика, но `pg_isready` не проверяет подключение к конкретной БД
23  build: . — убедитесь, что в директории есть Dockerfile, иначе билд упадёт
28  command: sh -c "alembic upgrade head ..." — можно разделить миграции и запуск в `entrypoint.sh` для читаемости и отладки
34  restart: "no" — логично для тестов, но можно добавить `depends_on` от `app`, если есть интеграционные тесты
39  pg_data: — volume не указан явно в `.dockerignore`, может потянуть мусор при билде

Yasuo20012001 commented on src/admin/auth.py:
```diff
@@ -0,0 +1,41 @@
+
+    async def authenticate(self, request: Request) -> bool:
+        """Authorize request based on is_superuser flag in session."""
+        return request.session.get("is_superuser", False)
```
9   class AdminAuth(AuthenticationBackend): — хорошее наследование от SQLAdmin backend, но стоит указать `__init__` если будут расширения
13  form = await request.form() — не обрабатывается возможная ошибка при парсинге формы (например, пустой `Content-Type`)
15  if not email or not password: — можно логировать попытку входа с отсутствующими данными для аудита
18  user = await session.scalar(...) — потенциальный `None` без логирования, можно добавить info-level лог
19  if not user or not verify_password(...) — объединённая проверка затрудняет отладку, лучше разделить для логирования причины
21  if not user.is_superuser: — пропускается возможность авторизации других ролей, если политика доступа изменится
23  request.session.update(...) — желательно использовать константы для ключей, чтобы избежать орфографических ошибок
28  return request.session.get("is_superuser", False) — можно добавить `type: ignore` если mypy ругается на динамическую сессию

Yasuo20012001 commented on src/admin/views.py:
```diff
@@ -0,0 +1,136 @@
+class TaskAssigneeAssociationAdmin(AssociationAdmin, model=TaskAssigneeAssociation):
+    """Admin view for task-assignee relationships."""
+    column_list = ["task.title", "user.email", "assigned_at"]
+    column_labels = {"task.title": "Задача", "user.email": "Исполнитель", "assigned_at": "Дата назначения"}
```
19  form_excluded_columns = ["password"] — скрытие пароля в форме корректно, но учти, что это не мешает редактированию через API напрямую
30  if not creator_id: — потенциальная точка отказа, можно логировать это для отладки сессии
32  team_in = TeamCreate(**data) — нет валидации структуры `data` перед сериализацией, может вызвать непонятную ошибку
52  if not creator_id or not team_id: — стоит разбить проверку для более точной диагностики проблемы (например, отсутствует user_id или team_id)
57  task_in = TaskCreate(**data) — аналогично, можно предварительно валидировать `data` для явных ошибок
72  column_formatters = {"changed_at": ...} — форматирование полезно, но может ломаться при `None` значении, лучше `getattr(..., '')`
79  can_create = False — запрет создания оправдан, но стоит документировать, почему записи истории недоступны для админов
91  AssociationAdmin — хорошее переиспользование шаблона для M2M-таблиц, но возможно стоит добавить `column_sortable_list`

Yasuo20012001 commented on src/deps/permissions.py:
```diff
@@ -0,0 +1,155 @@
+
+        return current_user
+
+    return verify
```
12  from src.schemas.user import UserPayload, TeamRole — импортирует `TeamRole`, но используется только в некоторых функциях, может быть отложен
39  if current_user.role != UserRole.ADMIN.value — возможно стоит использовать `Enum comparison` вместо `.value` для большей безопасности
60  if team_id not in [team.team_id for team in current_user.teams] — повторяется в нескольких функциях, можно вынести в утилиту
84  if team_assoc.role != TeamRole.MANAGER — возможен `AttributeError`, если структура `team_assoc` неконсистентна, лучше явно проверить
122  assignee_ids = [row[0] for row in result.all()] — использование индекса делает код хрупким, лучше распаковывать с именем
147  raise HTTPException(...) — хардкод строки "This endpoint is disabled", можно заменить на константу или исключение уровня приложения
165  stmt = select(model).where(getattr(model, "id") == resource_id) — потенциально опасно при отсутствии поля id, лучше валидировать заранее
180  user_result = await db.execute(user_stmt) — отдельный запрос к User ради `is_superuser`, хотя это поле можно включить в `UserPayload`
201  if getattr(resource, creator_field) != current_user.id — сообщение "Only the creator..." может быть вынесено в константы для переиспользования

AleksandrZaec committed:
feat(docker-compose.yaml):
- создал скрипт entrypoint.sh с миграциями Alembic и запуском Uvicorn
- изменён docker-compose.yml: заменён command на entrypoint с вызовом entrypoint.sh
- добавлено depends_on для тестового сервиса, чтобы тесты ждали запуска приложения и базы
- добавлен .dockerignore для исключения лишних файлов из контекста сборки
feat(src/admin/auth.py):
- настроено логирование через модуль logging с уровнями (info, warning, error)
- добавлен базовый конфиг логирования в main.py с читаемым форматом и таймстампом
- вынесены константы ключей сессии USER_ID и IS_SUPERUSER в auth.py
feat(src/admin/views.py):
- Добавлена валидация входных данных через Pydantic (UserCreate, TeamCreate, TaskCreate)
- Логируются ошибки сессии (creator_id, team_id) и ошибки валидации
- column_formatters защищены от None-значений
- Добавлены sortable-колонки для ассоциативных таблиц
- Обновлены комментарии и логика отказа по can_create

AleksandrZaec committed:
fetch(src/deps/permissions.py)
- отложенный импорт TeamRole
- использование Enum-сравнения для ролей
- добавлена утилита is_member_of_team
- защита от AttributeError для team_assoc
- улучшена работа с результатами SQL
- вынесены константы для сообщений и ошибок
- добавлена валидация поля id
